core/src/test/java/org/apache/calcite/rex/RexBuilderTest.java

```diff
@@ -44,13 +52,38 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotEquals;
 import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 /**
  * Test for {@link RexBuilder}.
  */
 public class RexBuilderTest {
 
+  private static final int PRECISION = 256;
+
+  /**
+   * MySqlTypeFactoryImpl provides a specific implementation of
+   * {@link SqlTypeFactoryImpl} which sets precision to 256 for VARCHAR.
+   */
+  private static class MySqlTypeFactoryImpl extends SqlTypeFactoryImpl {
+
+    MySqlTypeFactoryImpl(RelDataTypeSystem typeSystem) {
+      super(typeSystem);
+    }
+
+    @Override public RelDataType createTypeWithNullability(
+        final RelDataType type,
+        final boolean nullable) {
+      if (type.getSqlTypeName() == SqlTypeName.VARCHAR) {
```

Comment:
A better way to do this would be to subclass `RelDataTypeSystemImpl` and override `getDefaultPrecision` method which is meant exactly for this.  Then you can simply instantiate SqlTypeFactoryImpl with the custom type system.

Reply:
I am afraid it cannot work. The whole call stack is as follows:
```
RexCopier#copy 
    => RelDataTypeFactoryImpl#copy 
        => SqlTypeFactoryImpl#createTypeWithNullability
            => BasicSqlType#createWithNullability
                => new BasicSqlType(...)
```
You can see it does not use `getDefaultPrecision` in `RelDataTypeSystemImpl`.

Reply:
Indeed you are right, let's keep it as is!